llama_index/vector_stores/dynamodb.py

```diff
@@ -0,0 +1,144 @@
+        dynamodb_kvstore (DynamoDBKVStore): data store
+        namespace (Optional[str]): namespace
+    """
+
+    stores_text: bool = False
+
+    def __init__(
+        self, dynamodb_kvstore: DynamoDBKVStore, namespace: Optional[str] = None
+    ) -> None:
+        """Initialize params."""
+        self._kvstore = dynamodb_kvstore
+        namespace = namespace or DEFAULT_NAMESPACE
+        self._collection_embedding = f"{namespace}/embedding"
+        self._collection_text_id_to_doc_id = f"{namespace}/text_id_to_doc_id"
+        self._key_value = "value"
+
+    @classmethod
+    def from_table_name(
+        cls, table_name: str, namespace: Optional[str] = None
+    ) -> DynamoDBVectorStore:
+        """Load from DynamoDB table name."""
+        dynamodb_kvstore = DynamoDBKVStore.from_table_name(table_name=table_name)
+        return cls(dynamodb_kvstore=dynamodb_kvstore, namespace=namespace)
+
+    @property
+    def client(self) -> None:
+        """Get client."""
+        return None
+
+    def get(self, text_id: str) -> List[float]:
+        """Get embedding."""
+        item = self._kvstore.get(key=text_id, collection=self._collection_embedding)
+        item = cast(Dict[str, List[float]], item)
+        return item[self._key_value]
+
+    def add(self, embedding_results: List[NodeWithEmbedding]) -> List[str]:
+        """Add embedding_results to index."""
+        response = []
+        for result in embedding_results:
+            self._kvstore.put(
+                key=result.id,
+                val={self._key_value: result.embedding},
+                collection=self._collection_embedding,
+            )
+            self._kvstore.put(
+                key=result.id,
+                val={self._key_value: result.ref_doc_id},
+                collection=self._collection_text_id_to_doc_id,
+            )
+            response.append(result.id)
+        return response
+
+    def delete(self, doc_id: str, **delete_kwargs: Any) -> None:
+        """Delete a document."""
+        text_ids_to_delete = set()
+        for text_id, item in self._kvstore.get_all(
+            collection=self._collection_text_id_to_doc_id
+        ).items():
+            if doc_id == item[self._key_value]:
+                text_ids_to_delete.add(text_id)
+
+        for text_id in text_ids_to_delete:
+            self._kvstore.delete(key=text_id, collection=self._collection_embedding)
+            self._kvstore.delete(
+                key=text_id, collection=self._collection_text_id_to_doc_id
+            )
+
+    def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResult:
+        """Get nodes for response"""
+        if query.filters is not None:
+            raise ValueError(
+                "Metadata filters not implemented for SimpleVectorStore yet."
+            )
+
+        # TODO: consolidate with get_query_text_embedding_similarities
+        items = self._kvstore.get_all(collection=self._collection_embedding).items()
```

Comment:
just curious, what kind of latency should I expect on retrieving e.g. anywhere between hundreds to millions of embeddings from DynamoDB?

Reply:
Having the vector store retrieve all embeddings on every query seems a bit wasteful. Since DynamoDB doesn't support vector search API directly, I wonder if it's best to just support it as an option to "persist" an in-memory vector store into?

Reply:
We do not know what the latency will be.

My emphasis in this class is not to query, but to consider Index learning (adding nodes) as a priority.

You can also choose to persist in SimpleVectorStore, which is an in-memory vector store, but in that case you would have to convert to dict and then to JSON, etc., and then store it in storage. In this case, if there are a large number of vectors, the amount of memory may be insufficient.

We are considering the possibility of using DynamoDB as a vector store to reduce the amount of memory used by eliminating the need for JSON conversion of all the vectors at once.

Reply:
@Disiok

Reply:
Yep, though one thing I want to point out is that calling `get_all` here would pull all the vectors into memory as well, so won't be solving the memory limitation. 

But alas, I think this could work reasonably when the number of vectors is small. So I'm okay with landing it as is.